### ISSUE TITLE
Implement Display for error types

### ISSUE THREAD

**Issue by brendanzab** [gfx-rs/gfx#506]
@toqueteos [says](https://github.com/gfx-rs/gfx-rs/pull/505#issuecomment-71377572):

> `fmt::String` -> `fmt::Display`, this is required in order for `Result<T, TextureError>` to work. Otherwise `.unwrap` doesn't work. [Source](http://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap)

**Comment by bvssvni** [comment]
In order to print out the same as for the `Debug` trait, you can do something like this:

```Rust
impl fmt::Display for Error {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        use std::fmt::Debug;        

        self.fmt(f)
    }
}
```

**Comment by bvssvni** [comment]
I updated the example, because the old one leads to calling itself on latest Rust nightly.

**Comment by kvark** [comment]
Last bits are fixed by #1666
